test/functional/outputs/azuremonitor/forward_to_azuremonitor_test.go

```diff
@@ -0,0 +1,141 @@
+// go:build !fluentd
+package azuremonitor
+
+import (
+	"strings"
+	"time"
+
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/test/helpers/azuremonitor"
+	"github.com/openshift/cluster-logging-operator/test/helpers/rand"
+	v1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/framework/functional"
+)
+
+var _ = Describe("Forwarding to Azure Monitor Log ", func() {
+	const azureSecret = "azure-secret"
+	var (
+		framework  *functional.CollectorFunctionalFramework
+		mockServer *azuremonitor.MockServer
+		secret     *v1.Secret
+		sharedKey  = rand.Word(16)
+		inputName  = "custom-app"
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewCollectorFunctionalFrameworkUsingCollector(logging.LogCollectionTypeVector)
+		// Start mockServer server
+		mockServer = azuremonitor.NewMockServer(framework.Namespace, azuremonitor.Mockoon)
```

Comment:
Why can this test not rely upon the output container being added to the pod of the functional test?  It is just a container and there are benefits to all the containers running in the same pod.  If we really can not us the functional framework we should move this test to rely upon the e2e framework but I suspect its possible with the functional one

Reply:
The problem is: i need to know public URL of Route, before deploy collector, that's why i did it in this way. 
See my comment here: [forward_to_azuremonitor_test.go#L43](https://github.com/vparfonov/cluster-logging-operator/blob/ca8bfbe8c047156105447c6dc92289afce3d625c/test/functional/outputs/azuremonitor/forward_to_azuremonitor_test.go#L43).
I will try to find way how do it in one pod 

@jcantrill

Reply:
We do we need a public URL?  I thought there was a place to set the host?  Can we not craft the URL from "localhost" and a port?

Reply:
Another thing we could do is to modify the framework to mount  the `/etc/host` file into the collector container.  This would provide a way to resolve a name without needing a route

Reply:
Maybe we can do something with /etc/host, at the end URL must have format: https://<some-id>.<host>/

Reply:
merged to one pod